File: solutions/level_2_even_fibonacci_numbers/level_2.java
// filename: level_2.java
// Project Euler Level 2 solution
// https://projecteuler.net/problem=2
// Challenge: Find the sum of all even Fibonacci numbers not exceeding 4,000,000
// author: K0K0$H@ of GitHub
// last edit: 12/15/2025
// run on Linux/Mac terminal: javac level_2.java && java level_2
// run on Windows CMD/Powershell: javac level_2.java ; java level_2

import java.util.*;

public class level_2{

	static int sum = 0;
	static int a = 0;
	static int b = 1;
	static int c = 0;

	public static void main(String[] args){

		while(true){
			c = a;
			a = b;
			b = Fibonacci(c, b);
			if(b > 4000000){
				break;
			}
		}

		System.out.println(sum);
		return;
	}

	public static int Fibonacci(int x, int y){
		if((x+y)%2==0){
			sum = sum + x + y;
		}
		return(x+y);
	}

}

File: solutions/level_2_even_fibonacci_numbers/level_2.c
// filename: level_2.c
// solution to Project Euler Problem 2
// https://projecteuler.net/problem=2
// Challenge: find the sum of all even Fibonacci numbers not exceeding 4,000,000
#include <stdlib.h>
#include <stdio.h>

int main(){
	printf("%s\n", "Project Euler Level 2 challenge: find the sum of all even Fibonacci numbers not exceeding 4,000,000");
	
	int sum = 0;
	int a = 0;
	int b = 1;
	int c = 0;
	
	int Fibonacci(int x, int y){
		if((x+y)%2==0){
			sum = sum + (x+y);
		}
		return (x+y);
	}

	while(1 == 1){
		c = a; // saves previous a value before reassigning a
		a = b; // iterates a to progress Fibonacci sequence
		b = Fibonacci(c, b);
		if(b > 4000000){
			break;
		}
	}

	printf("%d\n", sum);
	return(0);
}

// by K0K0$H@ of GitHub
// last edit: 12/15/2025
